src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * @fileoverview Provides a D3js component to be used to draw a elevation
+ * profile chart.
+ *
+ * Example usage:
+ *
+ * var selection = d3.select('#element_id');
+ * var profile = ngeo.profile().hover(function(point) {
+ *   console.log(point.x, point.y);
+ * }).out(function() {console.log("out");});
+ * selection.datum(data).call(profile);
+ *
+ *
+ * The selection data to be used should look like the following:
+ *
+ * [
+ *     {
+ *         "y": 199340,
+ *         "values": {"mnt": 788.7},
+ *         "dist": 0.0,
+ *         "x": 541620
+ *     }, ...
+ * ]
+ *
+ */
+goog.provide('ngeo.profile');
+
+
+/**
+ *
+ * @return {Object}
+ * @export
+ *
+ */
+ngeo.profile = function() {
+
+  /**
+   * The values for margins around the chart defined in pixels.
+   */
+  var margin = {top: 10, right: 10, bottom: 30, left: 40};
```

Comment:
This would be great indeed, I agree that it would simplify the code. But did you give it a try?
How can we ensure that the axis labels are visible?